### ISSUE TITLE
2.4.4 regression for "Curiously Recurring Generics"

### ISSUE THREAD

**Issue by zfernandez88** [immutables/immutables#580]
Hello, after upgrading to 2.4.4, compiling the following code causes an error during compilation:
```Java
import org.immutables.value.Value;

@Value.Immutable
public interface Crtp<T extends Crtp<T>> {
}
```

Error trace:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project crtp-bug: Compilation failure
[ERROR] org.immutables.value.internal.$processor$.$Processor threw java.lang.IllegalArgumentException: no such type parameter 'T' defined in
[ERROR] at org.immutables.value.internal.$processor$.encode.$Type$Producer$1.variable($Type.java:397)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor$TypeConverter.visitTypeVariable($TypeExtractor.java:128)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor$TypeConverter.visitTypeVariable($TypeExtractor.java:95)
[ERROR] at com.sun.tools.javac.code.Type$TypeVar.accept(Type.java:1337)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor$TypeConverter.visitDeclared($TypeExtractor.java:116)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor$TypeConverter.visitDeclared($TypeExtractor.java:95)
[ERROR] at com.sun.tools.javac.code.Type$ClassType.accept(Type.java:944)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor.getBounds($TypeExtractor.java:86)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor.initParameters($TypeExtractor.java:75)
[ERROR] at org.immutables.value.internal.$processor$.encode.$TypeExtractor.<init>($TypeExtractor.java:46)
[ERROR] at org.immutables.value.internal.$processor$.meta.$ValueType.getTypeExtractor($ValueType.java:1553)
[ERROR] at org.immutables.value.internal.$processor$.meta.$ValueType.getBuildFromTypes($ValueType.java:1228)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables._t18__generateBuilder($Generator_Immutables.java:4812)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16115)
[ERROR] at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
[ERROR] at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables._t4__generateImmutable($Generator_Immutables.java:579)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16101)
[ERROR] at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
[ERROR] at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables$1.run($Generator_Immutables.java:23)
[ERROR] at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
[ERROR] at org.immutables.value.internal.$generator$.$Output$7.invoke($Output.java:163)
[ERROR] at org.immutables.value.internal.$generator$.$Intrinsics.$($Intrinsics.java:96)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables._t0__generate($Generator_Immutables.java:20)
[ERROR] at org.immutables.value.internal.$processor$.$Generator_Immutables$FragmentDispatch.run($Generator_Immutables.java:16097)
[ERROR] at org.immutables.value.internal.$generator$.$Templates$Fragment.invoke($Templates.java:248)
[ERROR] at org.immutables.value.internal.$generator$.$AbstractGenerator.invoke($AbstractGenerator.java:57)
[ERROR] at org.immutables.value.internal.$processor$.$Processor.process($Processor.java:63)
[ERROR] at org.immutables.value.internal.$generator$.$AbstractGenerator.process($AbstractGenerator.java:87)
[ERROR] at org.immutables.processor.ProxyProcessor.process(ProxyProcessor.java:72)
[ERROR] at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
```

**Comment by elucash** [comment]
the probable fix released in 2.4.6, please evaluate

**Comment by zfernandez88** [comment]
The same repro provided above fails with the same stack trace in 2.4.6
